app/resources/v1/street_images.js

```diff
@@ -17,70 +19,109 @@ exports.post = async function (req, res) {
-  const userId = req.userId
-
-  if (!userId) {
-    res.status(400).send('Please provide user ID.')
-    return
-  }
-
-  let user
-
-  try {
-    user = await User.findOne({ id: userId })
-  } catch (error) {
-    logger.error(error)
-    res.status(500).send('Error finding user.')
-  }
-
-  if (!user) {
-    res.status(404).send('User not found.')
-    return
-  }
-
-  // Is requesting user logged in?
-  if (user.login_tokens.indexOf(req.loginToken) === -1) {
-    res.status(401).end()
-    return
-  }
-
-  // 2) Check that street exists.
-  // 3) Verify that street is owned by logged in user.
+  // 1) Check if street exists.
   let street
 
   try {
     street = await Street.findOne({ id: req.params.street_id })
   } catch (error) {
     logger.error(error)
     res.status(500).send('Error finding street.')
+    return
   }
 
   if (!street) {
     res.status(400).send('Street not found.')
     return
-  } else if (street.creator_id.toString() !== user._id.toString()) {
-    res.status(404).send('Signed in user cannot upload street thumbnail.')
-    return
   }
 
-  // 4) Upload street thumbnail to cloudinary.
-  let response
+  // 2) Check if street thumbnail exists.
+  let resource
+  const publicId = `${config.env}/street_thumbnails/${street.id}`
 
   try {
-    const publicId = `${config.env}/street_thumbnails/${req.params.street_id}`
-    response = await cloudinary.v2.uploader.upload(image, { public_id: publicId })
+    resource = cloudinary.v2.api.resource(publicId)
   } catch (error) {
     logger.error(error)
-    res.status(500).send('Error uploading thumbnail to cloudinary.')
   }
 
-  if (!response) {
-    res.status(400).send('Could not upload thumbnail to cloudinary.')
-    return
+  const handleUploadStreetThumbnail = async function (publicId) {
+    if (!publicId) {
+      res.status(400).send('Please provide the public ID to be used.')
+      return
+    }
+
+    try {
+      resource = await cloudinary.v2.uploader.upload(image, { public_id: publicId })
+    } catch (error) {
+      logger.error(error)
+    }
+
+    if (!resource) {
+      res.status(500).send('Error uploading street thumbnail to Cloudinary.')
+      return
+    }
+
+    const thumbnail = {
+      public_id: resource.public_id,
+      width: resource.width,
+      height: resource.height,
+      format: resource.format,
+      secure_url: resource.secure_url,
+      created_at: resource.created_at
+    }
+
+    res.status(200).json(thumbnail)
   }
 
-  res.status(200).send('Successfully uploaded thumbnail to cloudinary.')
+  const handleFindStreetWithCreator = async function (street) {
+    if (!req.userId) {
+      res.status(404).send('Please provide a user ID.')
+      return
+    }
+
+    let user
+
+    try {
+      user = await User.findOne({ id: req.userId })
+    } catch (error) {
+      logger.error(error)
+      res.status(500).send('Error finding user.')
+      return
+    }
+
+    if (!user) {
+      res.status(400).send('User not found.')
```

Comment:
If this user is not found we should send `403 Forbidden` because they sent a user ID that isn't valid.